### PR TITLE
JAX: Add dropout.md

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
Note that dropout in flax requires passing rngs with a specific 'dropout' key. This is not required otherwise and is set to None by default. This PR handles that. In the future, we can refactor this further to make use of `make_rng` to make sure every dropout layer has a separate generation.

I also save a utility lambda function which can be used in later sections called `get_key`.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_multilayer-perceptrons/dropout.md [hunk 1]
```diff
@@ -358,12 +407,39 @@ class DropoutMLP(d2l.Classifier):
...
 Next, we [**train the model**].
 
 ```{.python .input}
 %%tab all
 model = DropoutMLP(**hparams)
-trainer.fit(model, data)
+if tab.selected('pytorch', 'mxnet', 'tensorflow'):
+    trainer.fit(model, data)
+if tab.selected('jax'):
+    trainer.fit(model, data, key={'params': get_key(), 'dropout': get_key()})
```
Comment (astonzhang) [inline chapter_multilayer-perceptrons/dropout.md:442]: Can we hide this `key` argument in the interface by scanning over the model to see if there exists dropout layers?

Comment (AnirudhDagar) [reply chapter_multilayer-perceptrons/dropout.md:442]: I'd lean for not doing that. I've mentioned this before, removing and hiding all these JAX/Flax intricacies is not right, and if we are adding JAX/Flax then why not do it the right way. I don't understand why are we so against these, there is nothing wrong with the key argument, it is used everywhere in Flax. In fact, we need to show that it is there.

Even if we hide it, it will only make the JAX version slower by scanning over the model to see if there exists dropout layers.

Comment (AnirudhDagar) [reply chapter_multilayer-perceptrons/dropout.md]: Ok, so based on our offline conversation, I've made a small hack to hide the key argument (making it optional).
`trainer.fit(model, data, key={'params': get_key(), 'dropout': get_key()})` -> `trainer.fit(model, data)`

We now pass the dropout key for all the models, and it will only be used by flax when a dropout layer is encountered. (not the most elegant, could also confuse a few readers. Hence I add a comment to explain).

I'm not a big fan of this approach about hiding the key, as i mentioned in our discussion, but it addresses your concerns about hiding the key, and for the readers, it looks simpler/readable. Also code across frameworks looks similar as requested.

Comment (astonzhang) [reply chapter_multilayer-perceptrons/dropout.md]: Technically, it is "nothing wrong" (make the code runnable) to have `trainer.fit(model, data, key)`. It is also "nothing wrong" to have something like `trainer.fit(model, data, num_layers)`. And it is also "nothing wrong" to keep adding arguments to `fit` as we keep having "special reasons" when growing the codebase. However, since `key` is the concern of `model`,  a more **scalable** design will recommend hiding `key` inside `model`.

I'm glad you found where it should go: yes, `loss` is part of the model :) I agree that there may be better ways, but given the cleaner (and more uniform) front-end in this version, let's stick to it for now and keep an eye on future improvement opportunities.
